### PR TITLE
Fix keys example in 1.9.x

### PR DESCRIPTION
**Pull request by laura-eprosima**

### CODE REVIEW

File: examples/C++/Keys/keys.cpp [hunk 1]
```diff
@@ -236,9 +253,10 @@ void keys()
     Subscriber* mySub = initSubscriber(sampleType, nullptr);
 
     // wait for the connection
-    while (pubListener.n_matched == 0)
+    if (pubListener.n_matched == 0)
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        std::unique_lock<std::mutex> lock(pubListener.mutex_);
+        pubListener.cv_.wait(lock);
```
Comment (LuisGP) [inline examples/C++/Keys/keys.cpp]: You can avoid the if by providing the CV a predicate: 

    pubListener.cv_.wait(lock, [&pubListener]()
        {
            return pubListener.n_matched > 0;
        });

File: examples/C++/Keys/keys.cpp [hunk 2]
```diff
@@ -300,9 +318,10 @@ void publisherKeys()
     Publisher* myPub = initPublisher(sampleType, pubListener);
 
     // wait for the connection
-    while (pubListener.n_matched == 0)
+    if (pubListener.n_matched == 0)
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        std::unique_lock<std::mutex> lock(pubListener.mutex_);
+        pubListener.cv_.wait(lock);
```
Comment (LuisGP) [inline examples/C++/Keys/keys.cpp]: Use a predicate instead of the *if*.

File: examples/C++/Keys/keys.cpp [hunk 3]
```diff
@@ -342,9 +361,10 @@ void subscriberKeys()
     initSubscriber(sampleType, &subListener);
 
     // wait for the connection
-    while (subListener.n_matched == 0)
+    if (subListener.n_matched == 0)
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        std::unique_lock<std::mutex> lock(subListener.mutex_);
+        subListener.cv_.wait(lock);
```
Comment (LuisGP) [inline examples/C++/Keys/keys.cpp]: Use a predicate instead of the *if*.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/649/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/563/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/645/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/127/)

**Pull request merged**